pkg/gather/collect/collect.go

```diff
@@ -0,0 +1,489 @@
+			resourceInfo.Resource.GroupResource().String(),
+		)
+
+	case meta.RESTScopeNameRoot:
+		return filepath.Join(
+			c.baseDir,
+			clusterScopedDirName,
+			resourceInfo.Resource.GroupResource().String(),
+		)
+
+	default:
+		return fmt.Sprintf("unrecognized scope %q", scope)
+	}
+}
+
+func (c *Collector) writeObject(ctx context.Context, dirPath string, obj kubeinterfaces.ObjectInterface, resourceInfo *ResourceInfo) error {
+	var err error
+	for _, printer := range c.printers {
+		filePath := filepath.Join(dirPath, obj.GetName()+printer.GetSuffix())
+		err = writeObject(printer, filePath, resourceInfo, obj)
+		if err != nil {
+			return fmt.Errorf("can't print object: %w", err)
+		}
+	}
+
+	return nil
+}
+
+func (c *Collector) writeResource(ctx context.Context, obj kubeinterfaces.ObjectInterface, resourceInfo *ResourceInfo) error {
+	resourceDir := c.getResourceDir(obj, resourceInfo)
+
+	err := os.MkdirAll(resourceDir, 0770)
+	if err != nil {
+		return fmt.Errorf("can't create resource dir %q: %w", resourceDir, err)
+	}
+
+	err = c.writeObject(ctx, resourceDir, obj, resourceInfo)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (c *Collector) collect(
+	ctx context.Context,
+	obj kubeinterfaces.ObjectInterface,
+	resourceInfo *ResourceInfo,
+) error {
+	err := c.writeResource(ctx, obj, resourceInfo)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func isPublicSecretKey(key string) bool {
+	switch key {
+	case "ca.crt", "tls.crt", "service-ca.crt":
+		return true
+	default:
+		return false
+	}
+}
+
+func (c *Collector) collectSecret(ctx context.Context, u *unstructured.Unstructured, resourceInfo *ResourceInfo) error {
+	secret := &corev1.Secret{}
+	err := runtime.DefaultUnstructuredConverter.FromUnstructured(u.Object, secret)
+	if err != nil {
+		return fmt.Errorf("can't convert secret from unstructured: %w", err)
+	}
+
+	for k := range secret.Data {
+		if !isPublicSecretKey(k) {
+			secret.Data[k] = []byte("<redacted>")
+		}
+	}
+
+	err = c.writeResource(ctx, secret, resourceInfo)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func retrieveContainerLogs(ctx context.Context, podClient corev1client.PodInterface, destinationPath string, podName string, logOptions *corev1.PodLogOptions) error {
+	dest, err := os.OpenFile(destinationPath, os.O_CREATE|os.O_WRONLY, 0666)
+	if err != nil {
+		return fmt.Errorf("can't open file %q: %w", destinationPath, err)
+	}
+	defer func() {
+		err := dest.Close()
+		if err != nil {
+			klog.ErrorS(err, "can't close file", "Path", destinationPath)
+		}
+	}()
+
+	logsReq := podClient.GetLogs(podName, logOptions)
+	readCloser, err := logsReq.Stream(ctx)
+	if err != nil {
+		return fmt.Errorf("can't create a log stream: %w", err)
+	}
+	defer func() {
+		err := readCloser.Close()
+		if err != nil {
+			klog.ErrorS(err, "can't close log stream", "Path", destinationPath, "Pod", podName, "Container", logOptions.Container)
+		}
+	}()
+
+	_, err = io.Copy(dest, readCloser)
```

Comment:
shouldn't this limit number of bytes collected? Seems command parameter is there but it's not used anywhere.

Reply:
no, but I should set it in PodLogOptions

Reply:
done in PodLogOptions